tool/tsh/tsh.go

```diff
@@ -485,6 +487,12 @@ func Run(args []string, opts ...cliOption) error {
 	proxyApp := proxy.Command("app", "Start local TLS proxy for app connection when using Teleport in single-port mode")
 	proxyApp.Arg("app", "The name of the application to start local proxy for").Required().StringVar(&cf.AppName)
 	proxyApp.Flag("port", "Specifies the source port used by by the proxy app listener").Short('p').StringVar(&cf.LocalProxyPort)
+	proxyAWS := proxy.Command("aws", "Start local HTTPS proxy for AWS access.")
+	proxyAWS.Flag("app", "Optional Name of the AWS application to use if logged into multiple.").StringVar(&cf.AppName)
+	proxyAWS.Flag("proxy-port", " Specifies the source port used by the HTTPS proxy listener.").Short('p').StringVar(&cf.LocalProxyPort)
+	proxyAWS.Flag("endpoint-url-port", " Specifies the port used for AWS endpoint URL.").Short('e').StringVar(&cf.AWSEndpointURLPort)
+	proxyAWS.Flag("verbose", "Show proxy information when executing a command.").Short('v').BoolVar(&cf.Verbose)
+	proxyAWS.Arg("command", "Command to execute after starting the local proxy.").StringsVar(&cf.AWSCommandArgs)
```

Comment:
I would personal incline to  `tsh aws --exec`  version but lets ask the product team: 

@xinding33 @klizhentas Could you help use with this UX dilemma ?

Brief overview: Right now we are having `the tsh aws` command that wraps the AWS native CLI. 
Stave in this PR added support for the local AWS proxy so a user can run `tsh proxy aws`  and use tools external tools like terraform but a user needs to set/export AWS envs variable before executing `terraform apply` so we want to add additional wrapper for an arbitrary AWS tool/bash command. 

Stave proposed following UX solutions: 
> 1. tsh proxy aws <command>
> 2. somehow put it in tsh aws maybe tsh aws --exec <command>
> 3. a new command like tsh aws exec <command>

Reply:
`tsh aws --exec <command>` is the easiest to implement too =D. I think I will go for that unless @xinding33 @klizhentas @r0mant have different ideas. Thanks

Reply:
@greedy52 @smallinsky can you describe in a bit more detail what would `tsh aws --exec` do differently from `tsh aws`. I am assuming the whole difference is that `tsh aws --exec` will create a local tunnel and run it? Can you describe how people would use it?

Reply:
TBH `tsh aws --exec` is a little confusing to me as well. If the issue is that a user needs to set additional environment variables, can `tsh proxy aws` just give them env vars to set? This would be consistent with how other `tsh proxy xxx` commands work.

Reply:
@klizhentas today `tsh aws <aws-cli-subcommand>` runs AWS CLI commands under our proxy, and this PR is adding the `tsh proxy aws` to support applications other than AWS CLI (say `terraform`).

However, to use `tsh proxy aws` like other `tsh proxy` commands, users have to
1. start `tsh proxy aws` proxy server, and keep it running
2. use the information dumped from `tsh proxy aws` to configure the other application, and run it separately

What I propose here is, if tsh can provide a shortcut to do 1 and 2 at the same time by letting tsh execute the application with all the required credentials as environment variables. **e.g `tsh aws --exec terraform` to run terraform directly under our proxy.**

The benefits are:
1. only one shell is required to run the application (and more friendly for scripting)
2. a lot less typing or copy/paste
3. hides details

@r0mant I personally find it very useful when testing the feature but I can remove if we feel it can be confusing.